hazelcast/src/main/java/com/hazelcast/internal/cluster/ClusterService.java

```diff
@@ -123,4 +126,41 @@
      */
     String getClusterId();
 
+    /**
+     * Changes the cluster version transactionally. Internally this method uses the same transaction infrastructure as
+     * {@link #changeClusterState(ClusterState)} and the transaction defaults are the same in this case as well
+     * ({@code TWO_PHASE} transaction with durability 1 by default).
+     *
+     * If the requested cluster version is same as the current one, nothing happens.
+     *
+     * If a node of the cluster is not compatible with the given cluster {@code version}, as implemented by
+     * {@link com.hazelcast.instance.NodeExtension#isNodeVersionCompatibleWith(Version)}, then a
+     * {@link com.hazelcast.internal.cluster.impl.VersionMismatchException} is thrown.
+     *
+     * If an invalid version transition is requested, for example changing to a different major version, an
+     * {@link IllegalArgumentException} is thrown.
+     *
+     * @param version new version of the cluster
+     * @since 3.8
+     */
```

Comment:
It's important to say here, that while the version is being changes the clusterState is changed to IN_TRANSITION and thus other nodes may not join. Did you verify this behavior?